### PR TITLE
javascript2/week1 homework

### PR DESCRIPTION
**Pull request by garyaldahoul**

### CODE REVIEW

File: javascript/javascript2/week1/index.js [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    if (wordsArray[i].length < shortestWordLength) {
+      shortestWordLength = wordsArray[i].length;
+      shortestWord = wordsArray[i];
+    }
+    */
+    wordsArray[i].length < shortestWordLength &&
+      (shortestWordLength = wordsArray[i].length) &&
+      (shortestWord = wordsArray[i]);
+  }
+  showShortestWord.innerHTML = `The Shortest Wrod is ${shortestWord} has ${shortestWordLength} letters.`;
```
Comment (Sreelatha-Vallabhaneni) [inline javascript/javascript2/week1/index.js]: Change the typo mistake in the word

File: javascript/javascript2/week1/index.js [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+function addWords() {
+  word = word.value;
+  words.push(word);
+}
+document
+  .getElementById("shortestWordButton")
+  .addEventListener("click", (words) => {
+    getShortestDanishWord(words);
+  });
```
Comment (Sreelatha-Vallabhaneni) [inline javascript/javascript2/week1/index.js]: Please delete commented code

File: javascript/javascript2/week1/index.js [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+  */
+function getShortestDanishWord(array) {
+  let shortestWord = "";
+  shortestWordLength = array[0].length;
```
Comment (Sreelatha-Vallabhaneni) [inline javascript/javascript2/week1/index.js:9]: I like the way of function and variable naming, it's more descriptive

File: javascript/javascript2/week1/index.js [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+  for (let i = 0; i < array.length; i++) {
+    /*
+    if (array[i].length < shortestWordLength) {
+      shortestWordLength = array[i].length;
+      shortestWord = array[i];
+    }
+    */
```
Comment (Sreelatha-Vallabhaneni) [inline javascript/javascript2/week1/index.js]: delete commented code

File: javascript/javascript2/week1/index.html [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+      <h1>Find And Count The Danish Letters</h1>
+
+      <input
+        id="danishText"
+        type="text"
+        name="danishText"
+        placeholder="Enter Danish Text,Please..."
+      />
+      <button id="countDanishLettersButton" onClick="countDanishLetters()">
+        Count The Dnish Letters
```
Comment (Sreelatha-Vallabhaneni) [inline javascript/javascript2/week1/index.html]: typo mistake in danish word

File: javascript/javascript2/week1/index.js [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+    } else if (
+      radioChoosen[i].checked &&
+      radioChoosen[i].value === "writeInField"
+    ) {
+      name.addEventListener("keyup", spiritAnimalNameGenerator);
+      spiritAnimalNameButton.removeEventListener(
+        "click",
+        spiritAnimalNameGenerator
+      );
+      name.removeEventListener("mouseover", spiritAnimalNameGenerator);
```
Comment (Sreelatha-Vallabhaneni) [inline javascript/javascript2/week1/index.js:143]: It's good to work on optional exercise / task

File: javascript/javascript2/week1/products/main.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
+console.log("Script loaded");
+
+const products = getAvailableProducts();
+
+function renderProducts(products) {
+  // your code here
+  const ul = document.querySelector("ul");
+  products.forEach((product) => {
```
Comment (Sreelatha-Vallabhaneni) [inline javascript/javascript2/week1/products/main.js:8]: perfect use of array method, and list rendered.....good work!

File: javascript/javascript2/week1/index.js [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+
+    newSpiritAnimalNames.splice(randomNumber, 1);
+
+    console.log(newSpiritAnimalNames);
+
+    console.log(spiritName);
+    return (showSpiritAnimalName.innerHTML = spiritName);
+  } else {
+    console.log("there is no more animls spirit names.");
+    showSpiritAnimalName.innerHTML = "there is no more animls spirit names.";
```
Comment (Sreelatha-Vallabhaneni) [inline javascript/javascript2/week1/index.js]: another typo mistake in animals

File: javascript/javascript2/week1/index.js [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+const newSpiritAnimalNames = [...spiritAnimals];
+function spiritAnimalNameGenerator() {
+  if (newSpiritAnimalNames.length > 0) {
+    const randomNumber = Math.floor(
+      Math.random() * newSpiritAnimalNames.length
+    );
+
+    const spiritName = `${name.value} ${newSpiritAnimalNames[randomNumber]}`;
+
+    newSpiritAnimalNames.splice(randomNumber, 1);
```
Comment (Sreelatha-Vallabhaneni) [inline javascript/javascript2/week1/index.js:102]: No need to use splice method, just delete this line you will get the new spirit animal continuously by clicking the button

Comment (garyaldahoul) [reply javascript/javascript2/week1/index.js:102]: Hi Sreelatha,
Thank you so much
I fix everything, you ask me to fix but I'll keep splice method in this task cause I don't to have the animal name again and I want to stop after the names inside the array finish

**Pull request merged**